### ISSUE TITLE
Is there any docs about using github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus in middleware v2

### ISSUE THREAD

**Issue by wrapupc** [grpc-ecosystem/go-grpc-middleware#558]
Api in github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus is quite different from https://github.com/grpc-ecosystem/go-grpc-prometheus, and i didn't find any guidance of how to use it. Is is still in progress?

**Comment by bwplotka** [comment]
No, it's ready to use. You can see example [here](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/examples/server/main.go#L140). Where would you expect the examples/documentation? 

I think it's fair to add migration guide to README, let's do that 👍🏽 

Thanks for feedback! Api should allow similar (and more) functionalities.

**Comment by abhimanyuchugh** [comment]
@bwplotka Do you know if the migration guide was added in the end? I can't find it anywhere.

Since the `go-grpc-prometheus` package used global vars/package imports to do metric registration, it means we can't support both old and new package to stagger the migration and need to do a complete cutover. Is it safe to assume the new package is a straight replacement and there are no breaking changes, otherwise it makes it difficult to test the migration to new package?

Thanks!
